typed-racket-lib/typed-racket/rep/rep-utils.rkt

```diff
@@ -36,19 +36,30 @@
   (and (list? l)
        (>= (length l) len)))
 
-(define-for-cond-contract name-ref/c
-  (or/c identifier?
-        (cons/c exact-integer? exact-integer?)))
+(define (name-ref/c x)
+  (or (identifier? x)
+      (and (pair? x)
+           (exact-integer? (car x))
+           (exact-integer? (cdr x)))))
 
 (define (name-ref=? x y)
   (cond
     [(identifier? x)
      (and (identifier? y) (free-identifier=? x y))]
     [else (equal? x y)]))
 
-(define id-table (make-free-id-table))
```

Comment:
we now "gensym" identifiers for temporary idents when working w/ refinements, etc -- we don't want them to hang around forever, so we use a weak-hash-table that uses free-identifier=? for comparisons